src/main/scala/is/hail/table/Table.scala

```diff
@@ -500,116 +464,29 @@ class Table(val hc: HailContext, val tir: TableIR) {
     copy(key = key.toArray[String])
   }
 
-  def select(exprs: Array[String]): Table = {
+  def select(expr: String): Table = {
     val ec = rowEvalContext()
+    val ast = Parser.parseToAST(expr, ec)
 
-    val (paths, types, f) = Parser.parseSelectExprs(exprs, ec)
-
-    val insertionPaths = paths.map {
-      case Left(name) => name
-      case Right(path) => path.last
-    }
-
-    val overlappingPaths = paths.counter.filter { case (n, i) => i != 1 }.keys
-
-    if (overlappingPaths.nonEmpty)
-      fatal(s"Found ${ overlappingPaths.size } ${ plural(overlappingPaths.size, "selected field name") } that are duplicated.\n" +
-        "Either rename manually or use the 'mangle' option to handle duplicates.\n Overlapping fields:\n  " +
-        s"@1", overlappingPaths.truncatable("\n  "))
-
-    val inserterBuilder = new ArrayBuilder[Inserter]()
-
-    val finalSignature = (insertionPaths, types).zipped.foldLeft(TStruct()) { case (vs, (p, sig)) =>
-      val (s: TStruct, i) = vs.insert(sig, p)
-      inserterBuilder += i
-      s
-    }
-
-    val inserters = inserterBuilder.result()
-
-    val annotF: Row => Row = { r =>
-      ec.set(1, r)
+    assert(ast.`type`.isInstanceOf[TStruct])
+    ast.typecheck(ec)
+    val structIR = ast.toIR()
 
-      f().zip(inserters)
-        .foldLeft(Row()) { case (a1, (v, inserter)) =>
-          inserter(a1, v).asInstanceOf[Row]
-        }
-    }
-
-    val newKey = key.filter(insertionPaths.toSet)
-
-    copy(rdd = rdd.map(annotF), signature = finalSignature, key = newKey)
-  }
-
-  def select(selectedColumns: String*): Table = select(selectedColumns.toArray)
-
-  def select(selectedColumns: java.util.ArrayList[String]): Table = select(selectedColumns.asScala.toArray)
-
-  def drop(columnsToDrop: Array[String]): Table = {
-    if (columnsToDrop.isEmpty)
-      return this
-
-    val fieldSet = fieldNames.toSet
-    assert(columnsToDrop.forall(fieldSet.contains))
-    val dropSet = columnsToDrop.toSet
-
-    val keepIndices = signature.fields
-      .filter(f => !dropSet.contains(f.name))
-      .map(_.index)
-
-    val newRowType = TStruct(keepIndices.map(signature.fields).map(f => (f.name, f.typ)): _*)
-    val localRowType = signature
-    val newRVD = rvd.mapPartitions(newRowType) { it =>
-
-      val rv2 = RegionValue()
-      val rvb = new RegionValueBuilder()
-
-      it.map { rv =>
-        rvb.set(rv.region)
-        rvb.start(newRowType)
-        rvb.startStruct()
-
-        var i = 0
-        while (i < keepIndices.length) {
-          rvb.addField(localRowType, rv, keepIndices(i))
-          i += 1
-        }
+    if (structIR.isDefined && globalSignature.size == 0) {
```

Comment:
I'd write this as:

```
structIR match {
  case Some(structIR) if globalSignature.size == 0 => new Table(...
  case _ => ...
}
```

Wait, wait, wait.  Why does this only work if there are no globals?  I don't see any reason for that.

Reply:
Not sure -- that's what the `annotate` code was doing, so I assumed it had to do with the global aggregators problem you sent me the email about.

Reply:
I think this was a hack I did because I was worried about copying the global annotations to a region for every row. This could be really bad for large globals, but we shoudl do it anyway.

Reply:
Ah.  Yeah, let's get rid of it.  It might be a problem for long pipelines with huge globals, but Dan's changes will address this in the next week or two.